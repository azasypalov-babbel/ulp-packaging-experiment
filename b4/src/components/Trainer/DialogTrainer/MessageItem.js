import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { rem } from 'polished';
import { markupStringToHTML } from '@lessonnine/babbel-markup-helper.js';

import Text from '../../shared/Text';

const activeStyles = css`
  background-color: ${(props) => props.theme.color.pastel.electric};
  border-color: ${(props) => props.theme.color.pastel.electric};
`;

const leftStyles = css`
  border-radius: 0 ${rem('12px')} ${rem('12px')} ${rem('12px')};
`;

const rightStyles = css`
  border-radius: ${rem('12px')} 0 ${rem('12px')} ${rem('12px')};
`;

const StyledWrapper = styled.div`
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: column;

  max-width: 100%;

  padding: ${rem('16px')};

  border: 2px solid ${(props) => props.theme.color.silver};
  background-color: ${(props) => props.theme.color.semantic.interactive.secondary.background};

  transition: border 0.1s linear, background-color 0.1s linear;

  & > * + * {
    margin-top: ${rem('8px')};
  }

  &:hover {
    border: 2px solid ${(props) => props.theme.color.semantic.interactive.secondary.border};

    ${({ active }) => active && activeStyles}
  }

  ${({ active }) => active && activeStyles}

  ${({ position }) => {
    if (position === 'left') return leftStyles;
    if (position === 'right') return rightStyles;
  }};
`;

const MessageItem = (props) => {
  const {
    children,
    secondaryText,
    active,
    position,
    onClick
  } = props;
  return (
    <StyledWrapper
      active={active}
      position={position}
      onClick={onClick}
    >
      <Text
        fontFamily="fontMilliardBook"
        color="spaceGray"
      >
        {children}
      </Text>
      { secondaryText &&
        <Text
          data-selector="item-translation"
          fontFamily="fontMilliardBook"
          color="spaceGrayW28"
          fontSize="small"
          dangerouslySetInnerHTML={{ __html: markupStringToHTML(secondaryText) }}
        />
      }
    </StyledWrapper>
  );
};

MessageItem.propTypes = {
  children: PropTypes.node.isRequired,
  secondaryText: PropTypes.string,
  active: PropTypes.bool,
  onClick: PropTypes.func,
  position: PropTypes.oneOf(['left', 'right']).isRequired
};

export default MessageItem;
