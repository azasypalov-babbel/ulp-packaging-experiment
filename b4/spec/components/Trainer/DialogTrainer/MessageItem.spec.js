import React from 'react';
import { shallow } from 'enzyme';
import MessageItem from '../../../../src/components/Trainer/DialogTrainer/MessageItem';

const defaultProps = {
  children: 'children',
  secondaryText: '"Y sho era" el **mejor**',
  active: true,
  onClick: jest.fn(),
  position: 'left'
};

describe('<MessageItem />', () => {
  test('it renders', () => {
    const wrapper = shallow(<MessageItem {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when secondary text exists', () => {
    test('renders formatted string', () => {
      const wrapper = shallow(<MessageItem {...defaultProps} />);
      const formattedText = wrapper
        .find('[data-selector="item-translation"]')
        .prop('dangerouslySetInnerHTML')['__html'];

      expect(formattedText).toEqual('<i>Y sho era</i> el <b>mejor</b>');
    });
  });

  describe('when secondary text does not exists', () => {
    test('does not render the text', () => {
      const props = {
        ...defaultProps,
        secondaryText: undefined
      };
      const wrapper = shallow(<MessageItem {...props} />);

      expect(wrapper.find('[data-selector="item-translation"]').exists()).toBeFalsy();
    });
  });
});

