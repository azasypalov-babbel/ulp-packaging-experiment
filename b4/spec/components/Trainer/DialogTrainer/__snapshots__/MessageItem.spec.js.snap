// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageItem /> it renders 1`] = `
<styled.div
  active={true}
  onClick={[MockFunction]}
  position="left"
>
  <styled.span
    color="spaceGray"
    fontFamily="fontMilliardBook"
  >
    children
  </styled.span>
  <styled.span
    color="spaceGrayW28"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<i>Y sho era</i> el <b>mejor</b>",
      }
    }
    data-selector="item-translation"
    fontFamily="fontMilliardBook"
    fontSize="small"
  />
</styled.div>
`;
